### ISSUE TITLE
Accessibility: Extend use of ARIA live regions

### ISSUE THREAD

**Issue by swissspidy** [GoogleForCreators/web-stories-wp#6059]
### Background

When a portion of a page is updated with JavaScript, the update is usually highlighted with animation and bright colors, and is easy to see. But if you don’t have the ability to see the screen, you don’t know this has happened, unless the updated region is marked as an ARIA live region.

If this isn’t marked, there’s no notification for screen readers. But it’s also possible that all the region content will be announced after an update, if the ARIA live region is too large. You want to provide users with just a simple, concise message.

That's what we have the [`useLiveRegion`](https://github.com/google/web-stories-wp/blob/9aad674ce32ea03cd4ce0363ff277625bee92633/assets/src/edit-story/utils/useLiveRegion.js) hook for. Sadly it's been a bit underused in the project, mostly due to lack of documentation and education. Time to change this!

This hook is a simple tool that creates and appends an ARIA live notifications area to the element where developers can dispatch text messages. Assistive technologies will automatically announce any text change in this area. This ARIA live region has an ARIA role of “status” so it has an implicit aria-live value of polite and an implicit aria-atomic value of true.

This means assistive technologies will notify users of updates but generally do not interrupt the current task, and updates take low priority. If you’re creating an application with higher priority updates (such as a notification that their current session is about to expire, for example), then you’ll want to create your own method with an aria-live value of assertive.

### This Task

This is a catch-all ticket to improve our usage of aforementioned hook.

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* ARIA live regions are used appropriately where needed to announce updates to screen readers

## Implementation Brief

* [ ] Move `useLiveRegion` to design system so it can be used by both editor and dashboard
* [ ] Document `useLiveRegion` why & how in accessibility docs
  (can copy some stuff from above for the intro)
* [ ] Snackbars: Add new `spokenMessage` prop that defaults to the snackbar message
* [ ] "Style" panel: if fiddling with colors announce `This color combination may be hard for people to read.`  
    NB: It looks like we don't have any text contrast checks in this panel, so maybe we cannot really  do this.
* [ ] Search results: announce `%d result found` / `%d results found` / `No results found`
* [ ] Auto complete dropdown: announce `%d results found, use up and down arrow keys to navigate` / ... / `No results found`
* [ ] ...more?

**Comment by csossi** [comment]
Not getting a "x results found" on Search:
![image.png](https://images.zenhubusercontent.com/5ea085e4704e72c0a99c6997/4a718343-92b8-4034-8497-7e72083501e5)

**Comment by samwhale** [comment]
@csossi thanks for checking this. I tried this locally and saw this when using voice over:
![gif](https://user-images.githubusercontent.com/22185279/116115015-2eb70f00-a677-11eb-9eb8-81883ede50a7.gif)


Seems like it's working for me. Can you give me a little more information about how it's not working for you?

**Comment by csossi** [comment]
Verified in QA
